facad1ng.py

```diff
@@ -57,61 +59,86 @@ def print_banners():
-        raise ValueError("Invalid custom domain. It should be just the domain name without 'http://' or '/'.")
+    domain_pattern = re.compile(
+        r'^([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,}$'
+    )
+
+    if not domain_pattern.match(domain):
+        raise ValueError("Invalid custom domain. Please provide a valid domain name.")
 
 def format_phish_keywords(keywords):
+    max_length = 15
+    if not isinstance(keywords, str):
+        raise TypeError("Input must be a string.")
+
+    if len(keywords) > max_length:
+        raise ValueError("Input string exceeds the maximum allowed length.")
+
     return "-".join(keywords.split())
 
 # Input from the user with validation
-while True:
-    web_url = input(f"{G}Enter the original link {W}(ex: https://www.ngrok.com): {W}")
-    try:
-        validate_web_url(web_url)
-        break
-    except ValueError as e:
-        print(e)
-
-while True:
-    custom_domain = input(f"\n{Y}Enter your custom domain {W}(ex: gmail.com): {W}")
-    try:
-        validate_custom_domain(custom_domain)
-        break
-    except ValueError as e:
-        print(e)
-
-while True:
-    phish = input(f"\n{C}Enter phishing keywords {W}(ex: free-stuff, login): {W}")
-    phish = format_phish_keywords(phish)
-    if " " not in phish:
-        break
-    else:
-        print("Phishing keywords should not contain spaces. Use '-' to separate them.")
-
-# Prepare the data for the request
-data = {
-    'url': web_url,
-    'shorturl': '',
-}
-
-# Shorten the original URL with multiple URL shorteners
-short_urls = [shortener.short(web_url) for shortener in shorteners]
-
-# Mask the URLs with custom domain and phishing keywords
-def mask_url(domain, keyword, url):
-    # Use urlparse to properly split the URL
-    parsed_url = urlparse(url)
-
-    # Reconstruct the URL with the custom domain and phishing keyword
-    return f"{parsed_url.scheme}://{domain}-{keyword}@{parsed_url.netloc}{parsed_url.path}"
-
-# Print the results
-print(f"\n{Y}Original URL:{W}", web_url, "\n")
-print(f"{G}[~] {R}Masked URL (using multiple shorteners):{W}")
-for i, short_url in enumerate(short_urls):
-    masked_url = mask_url(custom_domain, phish, short_url)
-    print(f"{G}╰➤ {C}Shortener {W} {i + 1}: {masked_url}")
+try:
+    while True:
+        web_url = input(f"{G}Enter the original link {W}(ex: https://www.ngrok.com): {W}")
+        try:
+            validate_web_url(web_url)
+            break
+        except ValueError as e:
+            print(e)
+
+    while True:
+        custom_domain = input(f"\n{Y}Enter your custom domain {W}(ex: gmail.com): {W}")
+        try:
+            validate_custom_domain(custom_domain)
+            break
+        except ValueError as e:
+            print(e)
+
+    while True:
+        phish = input(f"\n{C}Enter phishing keywords {W}(ex: free-stuff, login): {W}")
+        phish = format_phish_keywords(phish)
+        if " " not in phish:
+            break
```

Comment:
Don't use "break" here, we don't have to terminate the program. 
if user enters something wrong, then ask for the input again.

Reply:
that we can do.